### PR TITLE
Use GitHub Actions matrices for PHP and E2E.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue:

- #5030.

## Relevant technical choices

Included PHP version in the PHP matrices. They weren't explicitly mentioned in the ACs/IB, but it made sense and were probably intended to be included. 🤔

I also combined the lint and test files for easier organisation/scan-ability in actions, now they look like this:

CleanShot 2023-08-04 at 13 42 46

It's a bit easier to read through this way 🙂 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: .github/workflows/php-lint-tests.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      - 'composer.json'
+      - 'composer.lock'
+  pull_request:
+    branches:
+      - develop
+      - main
+      - 'feature/**'
+    # Only run if PHP-related files changed.
+    paths:
+      - '.github/workflows/php-tests-wp-latest-php-7-4.yml'
```
Comment (aaemnnosttv) [inline .github/workflows/php-lint-tests.yml:24]:
```suggestion
      - '.github/workflows/php-lint-tests.yml'
```

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @tofumatt – this is great to see so much consolidated. A few points to address but otherwise looking good!

**Comment by aaemnnosttv** [comment]
> I also combined the lint and test files for easier organisation/scan-ability in actions, now they look like this:

@tofumatt Let's leave the two separate workflows for now since they're concerned with some different files and we don't really benefit much from combining them in the same workflow. I initially thought that this was done to only run the PHP tests if the linting passed but that isn't the case. I also don't think that's necessary for the most part since we have linting as a pre-commit check, so it's probably rare that this would fail for us to justify delaying the other tests.

File: .github/workflows/php-lint-tests.yml [hunk 2]
```diff
@@ -0,0 +1,160 @@
+name: PHP Tests
+
+on:
+  push:
+    branches:
+      - develop
+      - main
+    # Only run if PHP-related files changed.
+    paths:
+      - '.github/workflows/php-tests.yml'
```
[the remark this answers is not in the document]
Comment (tofumatt) [reply .github/workflows/php-lint-tests.yml:10]: Argh, right on, I changed these file names a few times during development 🙂

File: .github/workflows/php-lint-tests.yml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+          extensions: mysqli, runkit, uopz
+          tools: composer:2.1
+          php-version: ${{ matrix.php_version }}
+        if: ${{ matrix.composer_php_version }}
+
+      - name: Update PHPUnit
+        run: |
+          composer config --unset platform.php
+          composer update phpunit/phpunit --with-dependencies --no-scripts
+        if: ${{ matrix.php_version == '8.0' }}
```
[the remark this answers is not in the document]
Comment (tofumatt) [reply .github/workflows/php-lint-tests.yml:150]: Yeah, I had some weird linting happen (with the Google, org-wide CodeQL tool) when I omitted them, you can see it in the commit history).

I'll try again, especially with the `startsWith`, as maybe it'll be okay then.

**Comment by tofumatt** [comment]
@aaemnnosttv My thinking was that it allows us to consolidate all PHP-related tests/workflows into one file and one set of `paths`. It could eventually allow us to share the build/install step too (or run `lint` as part of the test workflow, which would actually save us several minutes.)

The JS tests and linting are part of the same workflow AND job, this at least brings the PHP ones into the same workflow, and means one fewer set of _very similar_/identical paths to be maintained for PHP linting/testing.

We also certainly _could_ delay running tests until linting completes, but it's probably not worth it 🤔

File: .github/workflows/php-lint-tests.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (aaemnnosttv) [reply .github/workflows/php-lint-tests.yml]: Looks like it didn't like this line 🤷‍♂️ 
https://github.com/google/site-kit-wp/blob/9bb8ab1f7ba9888bcbc2b9d7014e4ef794e53d24/.github/workflows/php-tests.yml#L128

**Review by aaemnnosttv — APPROVED** [review]
Great!

**Pull request merged**